### PR TITLE
Fail bundles with killed dependencies if --allow-failed-dependencies not specified

### PR DESCRIPTION
**Pull request by bkgoksel**
Right now these bundles are left in `created` stage forever.

This PR has a large diff because of auto code formatting.

### CODE REVIEW

File: codalab/worker/bundle_manager.py [hunk 1]
```diff
@@ -472,11 +526,16 @@ def _construct_run_message(self, worker, bundle):
...
+        message['bundle'] = bundle_util.bundle_to_bundle_info(
+            self._model, bundle)
         if self._worker_model.shared_file_system and worker['user_id'] == self._model.root_user_id:
-            message['bundle']['location'] = self._bundle_store.get_bundle_location(bundle.uuid)
+            message['bundle'][
+                'location'] = self._bundle_store.get_bundle_location(
+                    bundle.uuid)
             for dependency in message['bundle']['dependencies']:
-                dependency['location'] = self._bundle_store.get_bundle_location(dependency['parent_uuid'])
+                dependency[
```
Comment (percyliang) [inline codalab/worker/bundle_manager.py]: This looks really ugly...can we not do this?

**Review by percyliang — COMMENTED** [review]
In general, can we separate the code formatting PRs from the functional changes?

**Comment by bkgoksel** [comment]
My bad, I installed an auto code formatter (https://github.com/google/yapf) to use for my other projects but forgot to turn it off while working on codalab. I can undo these changes and re-commit with just the semantic changes.

In general how do you feel about configuring an auto code formatter and generally enforcing it on all the commits? YAPF looks pretty customizable and we can try to configure it to avoid ugly changes like the earlier. It might help with long term consistency of code style. I think we can even add a CI check with it for PRs and everything. It also allows us to reformat the existing codebase while guaranteeing no functional changes will be made.

**Comment by percyliang** [comment]
I think it's a reasonable thing to do...we can just make a separate PR to
reformat all the code.  Maybe send one out so at least we can see what it
will look like.

**Pull request merged**